app/models/response.rb

```diff
@@ -1,48 +1,62 @@
-# A stateless response to a given set of questions.
-# Initializing is done by setting answer_values as a hash, for example:
-#   r = Response.new(answer_values: { 'EL02' => 1 })
-#
-# After this, all attributes can be read in an OOP way, for example:
-#   r.questions # All questions, including the next (unanswered) one
-#   # => [#<Question:0x007faadb3459a8 @id="EL02", @answer_value: 1>,
-#         #<Question:0x007faadb2d49d8 @id="EL03">]
+# A stateless response to a given set of questions in one ore more domains.
 class Response < BaseModel
   attr_accessor :uuid
 
   # Accessors for attributes defined by events
-  delegate :show_secret, to: :invitation_sent
+  delegate :show_secret, to: :invitation_accepted
   delegate :created_at, to: :invitation_accepted
   delegate :domain_ids, to: :invitation
+  delegate :requested_at, to: :invitation
 
-  # accessors for attributes defined by R package
-  %i( next_question_id estimate variance done estimate_interpretation warning ).each do |r_attribute|
-    define_method r_attribute do
-      ensure_valid do
-        RPackage.data_for(answer_values, domain_ids)[r_attribute]
-      end
+  def done
+    domain_responses.all?(&:done)
+  end
+
+  def domain_responses
+    domain_ids.map do |domain_id|
+      DomainResponse.find(uuid, domain_id)
+    end
+  end
+
+  # As stored in FinishResponse event
+  def results
+    domain_responses.map do |domain_response|
+      { answer_values: domain_response.answer_values,
+        estimate: domain_response.estimate,
+        variance: domain_response.variance,
+        estimate_interpretation: domain_response.estimate_interpretation,
+        warning: domain_response.warning }
     end
   end
 
+  def next_domain_response
```

Comment:
Ik heb hier dezelfde terminologie als voor de vragen aangehouden: wat is het domein van de volgende vraag. Niet: wat is het domein van de huidige vraag.
Volgens mij is dit in lijn met jou eerdere opzet en naamgeving.